BlueprintUI/Tests/ElementIdentifierTests.swift

```diff
@@ -9,58 +9,82 @@ class ElementIdentifierTests: XCTestCase {
         // Equal
 
         XCTAssertEqual(
-            ElementIdentifier(elementType: A.self, key: nil, count: 0),
-            ElementIdentifier(elementType: A.self, key: nil, count: 0)
+            ElementIdentifier.identifier(for: A(), key: nil, count: 0),
+            ElementIdentifier.identifier(for: A(), key: nil, count: 0)
         )
 
         XCTAssertEqual(
-            ElementIdentifier(elementType: A.self, key: "aKey", count: 0),
-            ElementIdentifier(elementType: A.self, key: "aKey", count: 0)
+            ElementIdentifier.identifier(for: A(), key: "aKey", count: 0),
+            ElementIdentifier.identifier(for: A(), key: "aKey", count: 0)
         )
 
         // Not Equal
 
         XCTAssertNotEqual(
-            ElementIdentifier(elementType: A.self, key: nil, count: 0),
-            ElementIdentifier(elementType: B.self, key: nil, count: 0)
+            ElementIdentifier.identifier(for: A(), key: nil, count: 0),
+            ElementIdentifier.identifier(for: B(), key: nil, count: 0)
         )
 
         XCTAssertNotEqual(
-            ElementIdentifier(elementType: A.self, key: nil, count: 0),
-            ElementIdentifier(elementType: A.self, key: nil, count: 1)
+            ElementIdentifier.identifier(for: A(), key: nil, count: 0),
+            ElementIdentifier.identifier(for: A(), key: nil, count: 1)
         )
 
         XCTAssertNotEqual(
-            ElementIdentifier(elementType: A.self, key: nil, count: 0),
-            ElementIdentifier(elementType: A.self, key: "aKey", count: 0)
+            ElementIdentifier.identifier(for: A(), key: nil, count: 0),
+            ElementIdentifier.identifier(for: A(), key: "aKey", count: 0)
         )
     }
+
+    func test_elementIdentifierCaching() {
+        let id1 = ElementIdentifier.identifier(for: A(), key: nil, count: 0)
+        let id2 = ElementIdentifier.identifier(for: B(), key: nil, count: 0)
+        let id3 = ElementIdentifier.identifier(for: A(), key: "unique", count: 0)
+        let id4 = ElementIdentifier.identifier(for: A(), key: nil, count: 1)
+        let id5 = ElementIdentifier.identifier(for: A(), key: nil, count: 0)
+        let id6 = ElementIdentifier.identifier(for: B(), key: nil, count: 0)
+
+        var set = Set([id1.objectIdentifier])
+        XCTAssertFalse(set.contains(id2.objectIdentifier))
+        set.insert(id2.objectIdentifier)
+        XCTAssertFalse(set.contains(id3.objectIdentifier))
+        set.insert(id3.objectIdentifier)
+        XCTAssertFalse(set.contains(id4.objectIdentifier))
+        set.insert(id4.objectIdentifier)
+        XCTAssertTrue(set.contains(id5.objectIdentifier))
+        set.insert(id5.objectIdentifier)
+        XCTAssertTrue(set.contains(id6.objectIdentifier))
+
+        XCTAssertEqual(id1.objectIdentifier, id5.objectIdentifier)
+        XCTAssertEqual(id2.objectIdentifier, id6.objectIdentifier)
+        XCTAssertNotEqual(id5.objectIdentifier, id6.objectIdentifier)
```

Comment:
Should we also check hashes now that `Hashable` and `Equatable` use different codepaths?